spark/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -17,141 +17,600 @@
+
+  var dfPart1: DataFrame = null
+  var dfPart2: DataFrame = null
+
+  var totalRows: Long = -1
+  var minId: Long = -1
+  var maxId: Long = -1
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    val df = spark.createDataFrame(Seq((1L, "a", 1L), (2L, "b", 1L), (3L, "c", 1L)))
-      .toDF("id", "data", "group")
-    val df2 = spark.createDataFrame(Seq(
-      (4L, "d", 1L),
-      (5L, "e", 1L),
-      (6L, "f", 1L),
-      (7L, null, 1L),
-      (8L, "b", 1L),
-      (9L, "b", 1L),
-      (10L, "b", 1L))).toDF("id", "data", "group")
+
+    dfPart1 = generateRowsDataFrame(spark.range(1L, 6L))
+    dfPart2 = generateRowsDataFrame(spark.range(6L, 11L))
 
     withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "false") {
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(noStatsTableName)
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(mixedStatsTableName)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(noStatsTableName)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(mixedStatsTableName)
 
       spark.sql(s"DELETE FROM $noStatsTableName WHERE id = 1")
       spark.sql(s"DELETE FROM $mixedStatsTableName WHERE id = 1")
 
-      df2.write.format("delta").mode("append").saveAsTable(noStatsTableName)
+      dfPart2.write.format("delta").mode("append").saveAsTable(noStatsTableName)
     }
 
     withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "true") {
-      import io.delta.tables._
-
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(testTableName)
-      df.write.format("delta").mode(SaveMode.Overwrite).save(testTablePath)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(testTableName)
 
       spark.sql(s"DELETE FROM $testTableName WHERE id = 1")
-      DeltaTable.forPath(spark, testTablePath).delete("id = 1")
 
-      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(testTableName)
-      df2.write.format("delta").mode(SaveMode.Append).save(testTablePath)
-      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(mixedStatsTableName)
+      dfPart2.write.format("delta").mode(SaveMode.Append).saveAsTable(testTableName)
+      dfPart2.write.format("delta").mode(SaveMode.Append).saveAsTable(mixedStatsTableName)
+
+      // Run updates to generate more Delta Log and trigger a checkpoint
+      // and make sure stats works after checkpoints
+      for (a <- 1 to 10) {
+        spark.sql(s"UPDATE $testTableName SET data='$a' WHERE id = 7")
+      }
+      spark.sql(s"UPDATE $testTableName SET data=NULL WHERE id = 7")
+
+      // Creates an empty (numRecords == 0) AddFile record
+      generateRowsDataFrame(spark.range(11L, 12L))
+        .write.format("delta").mode("append").saveAsTable(testTableName)
+      spark.sql(s"DELETE FROM $testTableName WHERE id = 11")
     }
+
+    val result = spark.sql(s"SELECT COUNT(*), MIN(id), MAX(id) FROM $testTableName").head
```

Comment:
@felipepessoto Can you make a change to collect these golden results without the optimization enabled? The optimization is enabled by default.

Reply:
Done